fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -155,41 +156,89 @@ private static int getFhirResourceFromBufferReader(BufferedReader resReader, int
     }
 
     public static int readFhirResourceFromObjectStore(AmazonS3 cosClient, String bucketName, String itemName,
-           int numOfLinesToSkip, List<Resource> fhirResources) {
-        int exported;
-        S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
-        try (S3ObjectInputStream s3InStream = item.getObjectContent();
-             BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream))) {
-            exported = getFhirResourceFromBufferReader(resReader, numOfLinesToSkip, fhirResources);
-            // Notify s3 client to abort and prevent the server from keeping on sending data.
-            s3InStream.abort();
-        } catch (Exception ioe) {
-            logger.warning("readFhirResourceFromObjectStore: " + "Error proccesing file " + itemName + " - " + ioe.getMessage());
-            exported = 0;
+           int numOfLinesToSkip, List<Resource> fhirResources, boolean isReuseInput, ImportTransientUserData transientUserData) {
+        int exported = 0;
+        if (isReuseInput) {
+            if (transientUserData.getBufferReader() == null) {
+                S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
+                S3ObjectInputStream s3InStream = item.getObjectContent();
+                transientUserData.setInputStream(s3InStream);
+                BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream));
+                transientUserData.setBufferReader(resReader);
+            }
+            try {
+                exported = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), 0, fhirResources);
+            } catch (Exception ioe) {
+                logger.warning("readFhirResourceFromObjectStore: " + "Error proccesing file " + itemName + " - " + ioe.getMessage());
+                exported = 0;
+            }
+
+        } else {
+            S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
```

Comment:
there seems like a high degree of repetition in the if(if)/else blocks, is there some opportunity to reduce the duplicated code (not the catch blocks are near identical)

Reply:
one are inside try() in which try take care of the close of the resources.